src/api/app/services/staging/staging_project_creator.rb

```diff
@@ -0,0 +1,45 @@
+module Staging
+  class StagingProjectCreator
+    def initialize(staging_project_names, staging_workflow)
+      @staging_project_names = staging_project_names
+      @staging_workflow = staging_workflow
+    end
+
+    def create
+      staging_projects = []
+      staging_project_names.each do |name|
+        project = Project.find_or_initialize_by(name: name)
```

Comment:
Is there any permission check that prevents users from creating projects they shouldn't be allowed to? :thinking:

Reply:
Nop, you are right!